docs/developers-guide/driver-changelog.md

```diff
@@ -0,0 +1,117 @@
+# Driver Interface Changelog
+
+Howdy! 👋
+
+Below is a list of changes to our driver interface for driver authors in each major version of Metabase.
+
+Generally, when we deprecate things, we'll list them here and wait three major versions before removing them entirely.
+
+## What's new for driver authors in Metabase 0.42.0
+
+Changes in Metabase 0.42.0 almost exclusively only affect drivers that derive from `:sql` (including `:sql-jdbc`).
+Non-SQL drivers likely will require no changes.
+
+0.42.0 introduces several significant changes to the way the SQL query processor compiles and determines aliases for
+MBQL `:field` clauses. For more background, see pull request
+[#19384](https://github.com/metabase/metabase/pull/19384).
+
+For the most part, SQL drivers don't need to make any changes unless they were doing manipulating Field or Table
+aliases. For drivers that were doing these things, lots of overlapping vars and methods have been consolidated, which
+means the bulk of the changes you'll need to make is simply deleting deprecated method implementations.
+
+#### Significant changes
+
+- The `metabase.driver.sql.query-processor/->honeysql` method for Field instances, e.g.
+
+  ```clj
+  (defmethod sql.qp/->honeysql [:my-driver (class Field)]
+    [driver field]
+    ...)
+  ```
+
+  is no longer invoked. All compilation is now handled by the MBQL `:field` clause method, e.g.
+
+  ```clj
+  (defmethod sql.qp/->honeysql [:my-driver :field]
+    [driver field-clause]
+    ...)
+  ```
+
+  If you were doing something special here, you'll need to move that special login into `[<driver> :field]` instead.
+  (You may no longer need this special logic, however -- see below.)
+
+- `:field`, `:expression`, and `:aggregation-options` clauses now contain information about what aliases you should
+  use to refer to them on both the left-hand side and right-hand side of SQL `AS` or elsewhere in a query. See PR
+  [#19610](https://github.com/metabase/metabase/pull/19610) for a detailed discussion about the new information,
+  hereafter referred to as the /[#19610 information](https://github.com/metabase/metabase/pull/19610)/.
+
+- `->honeysql` methods, specifically implementations for `:field`, should use or replace the
+  [#19610 information](https://github.com/metabase/metabase/pull/19610) rather than attempting to determine or
+  override it in some other fashion. If your driver does not have a custom implementation of `->honeysql` for `:field`
+  or `(class Field)`, you don't need to worry about this.
```

Comment:
```suggestion
- If you have a custom implementation of `->honeysql` for `:field` or `(class Field)`: `->honeysql` methods for `:field` should use or replace the
  [#19610 information](https://github.com/metabase/metabase/pull/19610) rather than attempting to determine or
  override it in some other fashion.
```